CMakeLists.txt

```diff
@@ -38,6 +38,24 @@ if(${enable_cbmc_tests})
     enable_testing()
 endif()
 
+find_package(Git)
+if(GIT_FOUND)
+  message(STATUS "Found Git executable: ${GIT_EXECUTABLE}")
+  execute_process(
+    COMMAND "${GIT_EXECUTABLE}" "describe" "--tags" "--always" "--long"
+    OUTPUT_VARIABLE git_repo_version
+    WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    )
```

Comment:
Does this actually happen on *every* build? I thought this would only happen on the initial CMake build directory setup operation, but not on subsequent `cmake --build` operations, which would mean it wouldn't pickup new commits/dirty working directory, etc.... I've not actually tried this though, so I may well be misunderstanding CMake here.

Reply:
I believe you are right. Let me try to figure out how to do it as needed, as is done for Makefiles.

Reply:
I'm actually out of ideas how to accomplish this, or at least how to do so in a portable fashion. I could invoke git via `add_custom_command`, but then the output needs to be written to a file. The contents of that file must be added to the compiler command line, which seems to be impossible. That leaves as workaround the non-portable option of adding `-include some-generated-file` to the compiler command-line. I might try this anyhow.

Reply:
Yeah, it seems Cmake is really unhelpful here :-( I guess another option is to use cmake's 'configure_file' in someway, but even that I think would somehow need to invoke a cmake configure.